core/src/main/java/ai/timefold/solver/core/api/solver/event/BestSolutionChangedEvent.java

```diff
@@ -58,6 +70,15 @@ public long getTimeMillisSpent() {
         return timeMillisSpent;
     }
 
+    /**
+     * @return A {@link EventProducerId} identifying what generated the event, either a
+     *         {@link SolveEventProducerId} if the cause is not associated with a Phase,
+     *         or {@link PhaseEventProducerId} if it is.
```

Comment:
Please switch the order - start with the case that doesn't require negation; the sentence becomes easier to read.